grass7/raster/r.valley.bottom/r.valley.bottom.py

```diff
@@ -106,536 +106,638 @@
 #%end
 
 
-from grass.pygrass.gis.region import Region
-from grass.pygrass.modules.shortcuts import general as g
-from grass.pygrass.modules.shortcuts import raster as r
-import sys
-import os
-import grass.script as grass
-import math
 import atexit
+import math
+import os
 import random
 import string
+import sys
+
+import grass.script as gs
+from grass.pygrass.gis.region import Region
+from grass.pygrass.modules.shortcuts import general as g
+from grass.pygrass.modules.shortcuts import raster as r
 
 if "GISBASE" not in os.environ:
     print("You must be in GRASS GIS to run this program.")
     sys.exit(1)
 
 
 def cleanup():
-    grass.message("Deleting intermediate files...")
-
+    """Clean-up procedure for module
+    """
+    gs.message("Deleting intermediate files...")
     for k, v in TMP_RAST.items():
         for f in v:
-            if len(grass.find_file(f)['fullname']) > 0:
-                grass.run_command(
-                    "g.remove", type="raster", name=f, flags="f", quiet=True)
+            if len(gs.find_file(f)["fullname"]) > 0:
+                g.remove(type="raster", name=f, flags="f", quiet=True)
+
+    current_region.write()
 
-    Region.write(current_region)
+
+def rand_id(prefix):
```

Comment:
Did you see: https://grass.osgeo.org/grass76/manuals/libpython/script.html#script.core.tempname

Reply:
Thanks Stefan,

In this case I'd like a custom name with a prefix because otherwise it's hard to debug r.valley.bottom, given the many processing steps with different outputs. That could be a potential enhancement for gs.tempname, i.e. the option of a prefix (like R's 'pattern') to generate something like "tmp_{prefix}_xxxxxx"?

Steve